Source/Scene/GroundPolygon.js

```diff
@@ -0,0 +1,424 @@
+/*global define*/
+define([
+        '../Core/BoundingSphere',
+        '../Core/Cartesian3',
+        '../Core/ComponentDatatype',
+        '../Core/defaultValue',
+        '../Core/defined',
+        '../Core/Ellipsoid',
+        '../Core/EllipsoidTangentPlane',
+        '../Core/EncodedCartesian3',
+        '../Core/IndexDatatype',
+        '../Core/Math',
+        '../Core/Matrix4',
+        '../Core/PolygonGeometryLibrary',
+        '../Core/PolygonPipeline',
+        '../Core/PrimitiveType',
+        '../Core/WindingOrder',
+        '../Renderer/BufferUsage',
+        '../Renderer/DrawCommand',
+        '../Renderer/ShaderSource',
+        '../Shaders/ShadowVolumeFS',
+        '../Shaders/ShadowVolumeVS',
+        './BlendingState',
+        './CullFace',
+        './Pass',
+        './StencilFunction',
+        './StencilOperation'
+    ], function(
+        BoundingSphere,
+        Cartesian3,
+        ComponentDatatype,
+        defaultValue,
+        defined,
+        Ellipsoid,
+        EllipsoidTangentPlane,
+        EncodedCartesian3,
+        IndexDatatype,
+        CesiumMath,
+        Matrix4,
+        PolygonGeometryLibrary,
+        PolygonPipeline,
+        PrimitiveType,
+        WindingOrder,
+        BufferUsage,
+        DrawCommand,
+        ShaderSource,
+        ShadowVolumeFS,
+        ShadowVolumeVS,
+        BlendingState,
+        CullFace,
+        Pass,
+        StencilFunction,
+        StencilOperation) {
+    "use strict";
+
+    var GroundPolygon = function(options) {
+        options = defaultValue(options, defaultValue.EMPTY_OBJECT);
+
+        var ellipsoid = defaultValue(options.ellipsoid, Ellipsoid.WGS84);
+        var granularity = defaultValue(options.granularity, CesiumMath.RADIANS_PER_DEGREE);
+        var polygonHierarchy = options.polygonHierarchy;
+
+        this._ellipsoid = ellipsoid;
+        this._granularity = granularity;
+        this._polygonHierarchy = polygonHierarchy;
+
+        this._boundingSphere = undefined;
+
+        this._va = undefined;
+        this._sp = undefined;
+        this._rs = undefined;
+
+        this.debugVolume = defaultValue(options.debugVolume, false);
+        this._debugVolumeCommand = undefined;
+
+        var that = this;
+        this._uniformMap = {
+            centralBodyMinimumAltitude : function() {
+                return -8500.0;
+            },
+            LODNegativeToleranceOverDistance : function() {
```

Comment:
Nevermind.  I see.  Perhaps the terrain geometric LOD can be an automatic uniform.